DPL Analysis: add RNTuple arrow::Dataset support

ktf opened the pull request:
As part of the changes, move the actual logic which serialises /
deserialised things to plugins so that we do not need to depend
or RNTuple in production code.

Include an initial converter to go from AO2Ds to RNTuple based files.

ktf commented:
Testing on hyperloop NOT ok. One leak fixed, one more leak to go.

alibuild commented:
Error while checking build/O2/fullCI for 8a0cef36211e08ddb87daa94cedb087876470bdf at 2024-12-12 13:48:
```
## sw/BUILD/O2-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/13761/8a0cef36211e08ddb87daa94cedb087876470bdf/build_O2_fullCI/pretty.html).

alibuild commented:
Error while checking build/O2/fullCI for 319d2cf6baa17cd623ed57323e714c726f618aec at 2024-12-12 14:11:
```
## sw/BUILD/O2-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'


## sw/BUILD/O2Physics-latest/log
/sw/SOURCES/O2Physics/13761-slc8_x86-64/0/Common/TableProducer/Converters/trackQA002Converter.cxx:25:32: error: 'tracksQA_000' was not declared in this scope; did you mean 'tracksQA_002'?
/sw/SOURCES/O2Physics/13761-slc8_x86-64/0/Common/TableProducer/Converters/trackQA002Converter.cxx:59:32: error: 'tracksQA_001' was not declared in this scope; did you mean 'tracksQA_002'?
ninja: build stopped: subcommand failed.
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/13761/319d2cf6baa17cd623ed57323e714c726f618aec/build_O2_fullCI/pretty.html).

ktf committed:
DPL Analysis: add RNTuple arrow::Dataset support

As part of the changes, move the actual logic which serialises /
deserialised things to plugins so that we do not need to depend
or RNTuple in production code.

Include an initial converter to go from AO2Ds to RNTuple based files.

alibuild commented:
Error while checking build/O2/fullCI for 12597a21796750aecde4dde78e99f7874aeb992b at 2024-12-12 14:58:
```
## sw/BUILD/O2-latest/log
c++: error: unrecognized command-line option '--rtlib=compiler-rt'
c++: error: unrecognized command-line option '--rtlib=compiler-rt'


## sw/BUILD/O2Physics-latest/log
/sw/SOURCES/O2Physics/13761-slc8_x86-64/0/Common/TableProducer/Converters/trackQA002Converter.cxx:25:32: error: 'tracksQA_000' was not declared in this scope; did you mean 'tracksQA_002'?
/sw/SOURCES/O2Physics/13761-slc8_x86-64/0/Common/TableProducer/Converters/trackQA002Converter.cxx:59:32: error: 'tracksQA_001' was not declared in this scope; did you mean 'tracksQA_002'?
ninja: build stopped: subcommand failed.


## sw/BUILD/o2checkcode-latest/log
--
========== List of errors found ==========
++ GRERR=0
++ grep -v clang-diagnostic-error error-log.txt
++ grep ' error:'
++ GRERR=1
++ [[ 1 == 0 ]]
++ mkdir -p /sw/INSTALLROOT/0b8f077854784a4701c27532820db902e340be6a/slc8_x86-64/o2checkcode/1.0-local759/etc/modulefiles
++ cat
--
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/13761/12597a21796750aecde4dde78e99f7874aeb992b/build_O2_fullCI/pretty.html).

ktf commented:
Orthogonal to everything else. Merging.
